L2DFileDialog/src/main.cpp

```diff
@@ -111,7 +111,7 @@ int main(int, char**)
 
 
         // L2DFileDialog code goes here.
-        static char* file_dialog_buffer = nullptr;
+        static char* file_dialog_buffer = new char[500];
```

Comment:
We don't need to allocate memory here because it's only used as a pointer to another buffer.

Reply:
The buffer is never initialized in the code, but rather just written into. This unsafe practice could corrupt the program. `strcpy_s` merely copies one buffer into another. It assumes that we allocated the memory pre-hand. Please take a look at the cppreferance below where the example shows the creation of a char buffer pre-hand.

https://en.cppreference.com/w/c/string/byte/strcpy

Please let me know if my understanding is flawed. I am open to discussion.

Reply:
Yes, if using `strcpy_s`, the buffer should be initialized beforehand. 

In this case, though, `file_dialog_buffer` simply points to a buffer that _is_ already initialized (`path1`, `path2`, or `path3`). As a result, although `file_dialog_buffer` is the pointer passed into the `L2DFileDialog`, it's pointing to an initialized block of memory and should be fine.

I haven't personally tried it with `strcpy_s` so my understanding could be wrong, but does what I said above make sense?

Reply:
If you're talking about the `path` variable on line 334 or line 345, then that is not the case. These variables are merely stack allocated. After they're copied into the `buffer` variable, they're popped off and thus their memory is freed which would corrupt the `buffer` variable if it was pointing to the `path` variable. `strcpy_s` copies the null-terminated string into the specified buffer, character by character. It does not move memory or change pointers.

I did not check the entirety of the codebase, only where the `buffer` variable is used. So if you used some cpp magic somewhere and therefore you're sure it's safe, I'll revert the allocation in the next commit.

Reply:
Risking memory safety to save 500 bytes isn't worth it. I'd need to dive deeper into the code than I have time for to confirm what I'm saying, so I'll trust your judgment here. I'll merge it now. Thanks for the discussion!

Reply:
If you find out through your future testing that what I'm saying is incorrect, do please let me know.